### ISSUE TITLE
What do "cased" and "uncased" mean?

### ISSUE THREAD

**Issue by 3ae248e4-2674-4fbe-8785-5e3c736a57cf** [python/cpython#88341]
BPO | [44175](https://bugs.python.org/issue44175)
--- | :---
Nosy | @rhettinger, @MojoVampire, @akulakov
PRs | <li>python/cpython#26865</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-05-19.03:45:40.004>
labels = ['3.9', 'docs']
title = 'What do "cased" and "uncased" mean?'
updated_at = <Date 2021-06-23.02:50:12.303>
user = 'https://bugs.python.org/otakutyrant'
```

bugs.python.org fields:
```python
activity = <Date 2021-06-23.02:50:12.303>
actor = 'andrei.avk'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation']
creation = <Date 2021-05-19.03:45:40.004>
creator = 'otakutyrant'
dependencies = []
files = []
hgrepos = []
issue_num = 44175
keywords = ['patch']
message_count = 8.0
messages = ['393920', '393921', '393954', '393983', '393989', '394098', '394099', '396378']
nosy_count = 5.0
nosy_names = ['rhettinger', 'docs@python', 'josh.r', 'andrei.avk', 'otakutyrant']
pr_nums = ['26865']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue44175'
versions = ['Python 3.9']
```

</p></details>

**Comment by andjc** [comment]
The current terminology of cased and uncased seems quite clear to me. Although references to Unicode general categories is potentially misleading. With the existing terminology, I suspect the wording is deliberate, it is drawing the distinction between pythons titlecasing and what Unicode defines as titlecasing.

If you want to see the differences in practical terms, I am using the title of an article from the Dinka Wikipedia:

```
>>> from icu import UnicodeString, Locale
>>> article = "Dɛ̈tëicëkäŋ akɔ̈ɔ̈n"
>>> print(article.title())
Dɛ̈Tëicëkäŋ Akɔ̈Ɔ̈N
>>> print(str(UnicodeString(article).toTitle(Locale.getRoot())))
Dɛ̈tëicëkäŋ Akɔ̈ɔ̈n
```

Basically, `str.title` works as design, it isn't a bug per se, rather an implementation decision take some point in the past. That said, I tend to use workarounds to `str.title`.

**Comment by StanFromIreland** [comment]
Some additional notes from the duplicate:


> The current documentation (Doc/library/stdtypes.rst) reads:
> 
> ```
> .. method:: str.istitle()
> 
>    Return ``True`` if the string is a titlecased string and there is at least one
>    character, for example uppercase characters may only follow uncased characters
>    and lowercase characters only cased ones.  Return ``False`` otherwise.
> ```
> 
> We could be more clear about what's happening:
> 
> - a titlecased string must contain at least one uppercase character
> - uppercase characters may not appear after any cased character (or: ... may appear only at the beginning of the string or following an uncased character)
> - lowercase characters may only follow cased characters
> - uncased characters may appear anywhere
> 
> [Caveat: I haven't checked these statements against the implementation.]
> 
> Other issues:
> - the current definition doesn't accurately describe the runtime behaviour for strings of uncased characters such as `"_"`: 
>   - at runtime `istitle()` returns `False`
>   - docs suggest this should return `True` (the string has at least one character and complies with the x-follows-y rules given)
> - we should probably say something about ligatures, especially titlecased code points such as "ǈ" (U+01C8)
>   - cursory testing shows these are accepted as if decomposed
> - could we reference a Unicode standard which describes our current behaviour for the edge cases like ligatures?
> 
> The documentation on `str.title()` is also silent on ligatures, and the behaviour isn't obvious (to me) when a titlecased variant of a cased code point is not available (e.g., "Æ/æ").
